src/util/mock-servers/service-testers/service-handlers/implementation/read-rows-handler.ts

```diff
@@ -0,0 +1,126 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// ** This file is automatically generated by gapic-generator-typescript. **
+// ** https://github.com/googleapis/gapic-generator-typescript **
+// ** All changes to this file may be overwritten. **
+
+import {SameCallHandler} from './same-call-handler';
+import {MockService} from '../../../mock-service';
+import {Mutation} from '../../../../../mutation';
+
+function rowResponse(rowKey: {}) {
+  return {
+    rowKey: Mutation.convertToBytes(rowKey),
+    familyName: {value: 'family'},
+    qualifier: {value: Mutation.convertToBytes('qualifier')},
+    valueSize: 0,
+    timestampMicros: 0,
+    labels: [],
+    commitRow: true,
+    value: Mutation.convertToBytes('value'),
+  };
+}
+
+export interface ReadRowsResponse {
+  row_keys?: string[];
+  last_row_key?: string;
+  end_with_error?: number;
+}
+
+export class ReadRowsHandler extends SameCallHandler {
+  responses: ReadRowsResponse[];
+  request: any = null;
+  callCount = 0;
+  message: any;
+
+  // TODO: service and endpoint should be bundled into one object.
+  constructor(
+    service: MockService,
+    endpoint: string,
+    responses: ReadRowsResponse[],
+    message?: any
+  ) {
+    super(service, endpoint);
+    this.responses = responses;
+    this.message = Object.assign({}, message);
+  }
+
+  // TODO: Create interface for this.
+  callHandler(call: any) {
```

Comment:
It would b good to use a better type for `call`, you could potentially just create your own interface:

```js
interface {
  emit: () => {}
}
```

But I wouldn't be surprised if there's a type available in the `grpc` library.